### DISCUSSION TITLE
TS Support gone for usePage hook?

### DISCUSSION THREAD

**Discussion by amiranagram** [inertiajs/inertia#1380] [Help (Vue)]
Hi! Before the latest (1.0.0) release I was able to type-hint the `usePage` hook, like this:

```ts
const { props }: { props: ComputedRef<SharedData> } = usePage()

const user = computed(() => props.value.auth.user)
```

I was hoping that after ditching the computed props in `usePage` I would be able to do this:

```ts
const { props }: { props: SharedData } = usePage()

const user = computed(() => props.auth.user)
```

but not anymore.

NOTE that `SharedData` is my own type that I wrote for everything I'm passing from the inertia middleware.

**Comment by talelmishali** [comment]
> Hi! Before the latest (1.0.0) release I was able to type-hint the `usePage` hook, like this:
> 
> 
> 
> ```ts
> 
> const { props }: { props: ComputedRef<SharedData> } = usePage()
> 
> 
> 
> const user = computed(() => props.value.auth.user)
> 
> ```
> 
> 
> 
> I was hoping that after ditching the computed props in `usePage` I would be able to do this:
> 
> 
> 
> 
> 
> ```ts
> 
> const { props }: { props: SharedData } = usePage()
> 
> 
> 
> const user = computed(() => props.auth.user)
> 
> ```
> 
> 
> 
> but not anymore.
> 
> 
> 
> NOTE that `SharedData` is my own type that I wrote for everything I'm passing from the inertia middleware.

It looks like it's in the work https://github.com/inertiajs/inertia/commit/25c56b7fdc28870b0b384d36eb6e2d2d4865b083

EDIT: 

Following @lepikhinb suggestion on the Discord channel, you can declare the ``usePage`` yourself. [Discord Comment](https://discord.com/channels/592327939920494592/592328250923941898/1062876743251267685)

```js
import "@inertiajs/vue3"
import { Page } from "@inertiajs/core"

declare module "@inertiajs/vue3" {
  export declare function usePage(): Page<{Your.Type}>
}
```
